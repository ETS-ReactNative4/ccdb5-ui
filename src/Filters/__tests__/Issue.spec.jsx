import React from 'react'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux'
import ReduxIssue, { Issue } from '../Issue'

function setupSnapshot(initial) {
  const middlewares = [thunk]
  const mockStore = configureMockStore(middlewares)
  const store = mockStore({
    query: {},
    aggs: {
      issue: initial
    }
  })

  return renderer.create(
    <Provider store={store}>
      <IntlProvider locale="en">
        <ReduxIssue />
      </IntlProvider>
    </Provider>
  )
}

describe('component:Issue', () => {
  describe('snapshots', () => {
    it('renders without crashing', () => {
      const target = setupSnapshot([])
      let tree = target.toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('only shows the first five items', () => {
      const options = [
        {
          "sub_issue.raw": {
            "buckets": [
              {"key":"Information is not mine","doc_count":7030},
              {"key":"Account status","doc_count":5103}
            ]
          },
          "key": "Incorrect information on credit report",
          "doc_count": 20991
        },
        {
          "sub_issue.raw": {
            "buckets": [
              {"key":"Debt is not mine","doc_count":9990},
              {"key":"Debt was paid","doc_count":4632}
            ],
          },
          "key": "Cont'd attempts collect debt not owed",
          "doc_count": 17244
        },
        {
          "sub_issue.raw": {
            "buckets": [],
          },
          "key": "Loan servicing, payments, escrow account",
          "doc_count": 14605
        },
        {
          "sub_issue.raw": {
            "buckets": [],
          },
          "key": "Loan modification,collection,foreclosure",
          "doc_count": 10716
        },
        {
          "sub_issue.raw": {
            "buckets": [],
          },
          "key": "Dealing with my lender or servicer",
          "doc_count": 7783
        },
        {
          "sub_issue.raw": {
            "buckets": [],
          },
          "key": "Not here",
          "doc_count": 999
        }
      ]

      const target = setupSnapshot(options)
      let tree = target.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})