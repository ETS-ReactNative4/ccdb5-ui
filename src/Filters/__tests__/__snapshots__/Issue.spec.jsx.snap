exports[`component:Issue snapshots only shows the first five items 1`] = `
<section
  className="collapsible-filter aggregation">
  <div
    className="layout-row">
    <h5
      className="flex-all">
      Issue / sub-issue
    </h5>
    <div
      className="flex-fixed toggle">
      <button
        className="a-btn a-btn__link hover"
        onClick={[Function]}>
        Hide
        <span
          className="cf-icon cf-icon-minus-round" />
      </button>
    </div>
  </div>
  <p>
    The type of issue and sub-issue the consumer identified in the complaint
  </p>
  <ul>
    <div
      className="aggregation-branch">
      <li
        className="flex-fixed layout-row parent">
        <input
          aria-label="Incorrect information on credit report"
          checked={false}
          className="flex-fixed"
          onClick={[Function]}
          type="checkbox" />
        <div
          className="flex-all toggle">
          <button
            className="a-btn a-btn__link hover"
            onClick={[Function]}>
            <span>
              Incorrect information on credit report
            </span>
            <span
              className="cf-icon cf-icon-down" />
          </button>
        </div>
        <span
          className="flex-fixed parent-count">
          <span>
            20,991
          </span>
        </span>
      </li>
    </div>
    <div
      className="aggregation-branch">
      <li
        className="flex-fixed layout-row parent">
        <input
          aria-label="Cont\'d attempts collect debt not owed"
          checked={false}
          className="flex-fixed"
          onClick={[Function]}
          type="checkbox" />
        <div
          className="flex-all toggle">
          <button
            className="a-btn a-btn__link hover"
            onClick={[Function]}>
            <span>
              Cont\'d attempts collect debt not owed
            </span>
            <span
              className="cf-icon cf-icon-down" />
          </button>
        </div>
        <span
          className="flex-fixed parent-count">
          <span>
            17,244
          </span>
        </span>
      </li>
    </div>
    <li
      className="flex-fixed layout-row">
      <input
        aria-label="Loan servicing, payments, escrow account"
        checked={false}
        className="flex-fixed"
        onClick={[Function]}
        type="checkbox" />
      <span
        className="flex-all bucket-key">
        Loan servicing, payments, escrow account
      </span>
      <span
        className="flex-fixed bucket-count">
        <span>
          14,605
        </span>
      </span>
    </li>
    <li
      className="flex-fixed layout-row">
      <input
        aria-label="Loan modification,collection,foreclosure"
        checked={false}
        className="flex-fixed"
        onClick={[Function]}
        type="checkbox" />
      <span
        className="flex-all bucket-key">
        Loan modification,collection,foreclosure
      </span>
      <span
        className="flex-fixed bucket-count">
        <span>
          10,716
        </span>
      </span>
    </li>
    <li
      className="flex-fixed layout-row">
      <input
        aria-label="Dealing with my lender or servicer"
        checked={false}
        className="flex-fixed"
        onClick={[Function]}
        type="checkbox" />
      <span
        className="flex-all bucket-key">
        Dealing with my lender or servicer
      </span>
      <span
        className="flex-fixed bucket-count">
        <span>
          7,783
        </span>
      </span>
    </li>
  </ul>
  <div
    className="flex-fixed">
    <button
      className="a-btn a-btn__link hover more">
      + Show 
      1
       more
    </button>
  </div>
</section>
`;

exports[`component:Issue snapshots renders without crashing 1`] = `
<section
  className="collapsible-filter aggregation">
  <div
    className="layout-row">
    <h5
      className="flex-all">
      Issue / sub-issue
    </h5>
    <div
      className="flex-fixed toggle">
      <button
        className="a-btn a-btn__link hover"
        onClick={[Function]}>
        Hide
        <span
          className="cf-icon cf-icon-minus-round" />
      </button>
    </div>
  </div>
  <p>
    The type of issue and sub-issue the consumer identified in the complaint
  </p>
  <ul />
</section>
`;
